File: Week_01_cpp/21.cpp

struct ListNode 
{
    int value;
    ListNode * next;
};


ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
 
 
 ListNode* h = new(ListNode);
 ListNode* cur = h;
 

 do {
    if (l1->value <= l2->value) {
        //插入q
        cur->next = l1; 
        l1 = l1->next;
    } else {
        cur->next = l2;
        l2 = l2->next;
    }
    cur = cur->next;

 }while(l1 != nullptr && l2 != nullptr)

if (l1 != nullptr) {
    cur->next = l1;
}

if (l2 != nullptr) {
    cur->next = l2;
}

return h->next;
}

File: Week_01/21.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func mergeTwoLists(l1 *ListNode, l2 *ListNode) *ListNode {

	head := &ListNode{}
	cur := head

	for l1.Next != nil && l2.Next != nil {
		if l1.Val <= l2.Val {
			cur.Next = l1
			l1 = l1.Next
		} else {
			cur.Next = l2
			l2 = l2.Next
		}
		cur = cur.Next
	}

	if l1.Next == nil {
		cur.Next = l2
	}
	if l2.Next == nil {
		cur.Next = l1
	}

	return head.Next
}